src/hotspot/cpu/zero/zeroInterpreter_zero.cpp

```diff
@@ -37,6 +37,7 @@
 #include "oops/method.hpp"
 #include "oops/oop.inline.hpp"
 #include "prims/jvmtiExport.hpp"
+#include "runtime/arguments.hpp"
```

Comment:
Why? `arguments.hpp` transitively includes it.

Reply:
> Why? `arguments.hpp` transitively includes it.

I thought the rule is not to rely on transitive includes. Unless it is an explicit umbrella header. If you need it, include it - that way if someone changes the include tree, we minimize the possibility of seemingly unrelated build errors.

Reply:
Ah, you mean because we use `RewriteBytecodes`, I see. Yes, we need the explicit header. My additional patch uses `UseSharedSpaces`, so it needs the additional include too.

Reply:
Yes, RewriteBytecode. Sorry, should have said.